string[] properties
string[] values
---
bool result
string instance
string class_name